### ISSUE TITLE
Add new cloud event for provider health check status change

### ISSUE THREAD

**Issue by autodidaddict** [wasmCloud/wasmcloud-otp#519]
Right now there's no way to remotely monitor whether a given provider has had its health check state go from pass to fail or from fail to pass. To avoid spamming, we shouldn't emit duplicate events in succession if the provider is in the same state as the last health check.

When a provider switches from passing a health check to failing, it should emit a `provider_health_check_failed` cloud event on the `wasmbus.evt.{x}` topic. Conversely, if a provider passes a health check if the previous status was a fail, then it should emit a `provider_health_check_passed` cloud event.

The `ProviderModule` `GenServer` is probably a good place to stick a `last_check` result state field that can be used to compare after performing the RPC-based health check over NATS.

This issue stems from a discussion with @jordan-rash , CC @giusdp .

**Comment by jordan-rash** [comment]
The Healthcheck contract has two fields.  If we are only emitting changes (true/false) in the `healthy` boolean, we are potentially missing a lot of good context in the `message`.  For example, I could have something thats healthy, but some state is deteriorated.  I don't want to mark the asset unhealthy, but I do what to be notified.  Currently, we request health every 30 seconds, I dont think it would be too noisy to just emit the status with message every request with a event of `provider_health_check_status` or something similar????

```smithy
/// Return value from actors and providers for health check status
@wasmbusData
structure HealthCheckResponse {
  /// A flag that indicates the the actor is healthy
  @n(0)
  healthy: Boolean
  
  /// A message containing additional information about the actors health
  @n(1)
  message: String
}
```

**Comment by autodidaddict** [comment]
It's 30 seconds per running provider. That's probably not that bad, considering the host emits its own heartbeat every 30 seconds and there's rarely more than 10 providers on a host

**Comment by giusdp** [comment]
I'm slowly learning the host_core codebase. I saw the providers are GenServers (managed by a provider supervisor) and they handle a `:do_health` message. I saw this snippet in that function:

 ```elixir
   # Only publish health check pass/fail when state changes
    state =
      case res do
        {:ok, _body} when not state.healthy ->
          publish_health_passed(state)
          %State{state | healthy: true}

        {:ok, _body} ->
          state

        {:error, _} ->
          if state.healthy do
            publish_health_failed(state)
            %State{state | healthy: false}
          else
            state
          end
      end  
```

Is it not already sending the health updates **only** when state actually changes?

**Comment by jordan-rash** [comment]
It does send messages to the RPC bus, (more context [here](https://github.com/wasmCloud/wasmcloud-otp/issues/517))...but it doesnt emit an event to `wasmbus.evt` stream, which is the appropriate place to read the messages, not the RPC stream

**Comment by giusdp** [comment]
I was checking this again. Maybe I should write a provider that I can put in healthy/unhealthy status on command to do some tests... 

What I'm confused about is that I do receive the `health_check_passed` event on `wasmbus.evt.default` when deploying a provider (the first  health check pass event). So I'm not really understanding what should be changed.

**Comment by giusdp** [comment]
Maybe this can be closed now that #560 was merged.
